sunpy/net/helioviewer.py

```diff
@@ -450,3 +499,18 @@ def _request(self, params):
     def _format_date(self, date):
         """Formats a date for Helioviewer API requests"""
         return parse_time(date).isot + "Z"
+
+    def _get_source_id(self, source_id, key):
+        if source_id is None:
+            try:
+                source_id_list = list(partial_match(key,self.data_sources))
+                if len(source_id_list) == 1:
+                    source_id = source_id_list[0]
+                else:
+                    raise KeyError("The values used for observatory, instrument, detector, measurement "
+                                   "do not correspond to a source_id. Please check the list using "
+                                   "HelioviewerClient.data_sources. Expected 1 got " + str(len(source_id_list)) + 
+                                   " values for source_id ", source_id_list)
```

Comment:
Have you seen how the error message is printed out in a terminal?

Reply:
I will upload a screenshot

Reply:
![error](https://user-images.githubusercontent.com/31286078/52524273-391ca680-2cc1-11e9-9ffb-490e59657926.png)

Reply:
The error message doesn't make it clear to the user what keys were returned.

Reply:
Ok I will rephrase it